### ISSUE TITLE
Remove resources from non-legacy policies [ui only]

### ISSUE THREAD

**Issue by blakestier** [chef/automate#3331]
Since users cannot define resources on their policies, the `*` that they see when viewing policies is confusing.

We attempted to do this once before but was perhaps complicated as we supported v1 at the time?

We should remove resources wherever possible.

### from sprint planning
- remove resources from non-legacy policies in the ui

**Comment by bcmdarroch** [comment]
This might be worth doing later, once we can confirm more customers

The problem with removing resources is that it makes legacy policies look like they have more permissions than they actually have.

This v1 default policy, which grants any action on compliance resources:
```
id: "some_guid"
subjects: [ "all:users"]
action: "*"
resource: "compliance:*"
```

will look like an all-access policy without its resource:
```
id: "(legacy) some_guid"
subjects: [ "all:users"]
actions: ["*"]
```

The solution might be to do a migration at the same time we drop resources, so that policy's resource would become its action 🤔 Not sure if that'll work for every policy though

**Comment by susanev** [comment]
is it possible to only hide resources on non-legacy policies?

**Comment by srenatus** [comment]
When discussing this in sprint planning, we've considered various ways to remove resources from the API while maintaining backwards-compatibility and a way to non-confusingly display legacy policies.

Some facts

- We cannot remove the `resources` field from the response messages because of back-compat. If we did that (not caring about compat), we'd still display (on the API level) confusing legacy policies as @bcmdarroch indicated above.
- @blakestier proposed to keep the `resources` field untouched (while ignoring it in the UI), and introduce a `legacyResources` field that would only be populated for legacy policies, and `""` for all others. Depending on how our API is used, the default fields (`""` for string) may or may not be displayed. (See for example `--emit-defaults` in `grpcurl`.) For our JSON API, **it would be displayed**, because we've configured it to emit defaults -- otherwise, the default enum value for some of our APIs would not be displayed, causing even more confusion. (If we could, changing those enums to have an `WHATEVER_UNSET` as first value would be preferrable.)
- If we don't care about back-compat for these, we could return a nested message, with a `oneof` field carrying either a `Policy` or a `LegacyPolicy`, where the former would *not* have a `resources` field. This would imply a few changes to our UI code and probably many changes to our API tests. (:warning: I haven't verified this.)
- Of course we could also introduce `/apis/iam/v3beta` 😉 (I'll show myself out.)
